Update CGAL to version 4.14

lrineau opened the pull request:
<s>Do not merge yet!</s>

This pull-request is a test to update the CGAL port to version 4.14-beta2. That CGAL version is currently a beta version. I will update the port once the official release it out.

Fix #5888.

lrineau committed:
CGAL-4.14-beta2

Rastaban commented on ports/cgal/portfile.cmake:
```diff
@@ -3,11 +3,16 @@ include(vcpkg_common_functions)
     HEAD_REF master
 )
 
+vcpkg_apply_patches(
```
This can be done with the PATCHES argument to vcpkg_from_github

Rastaban commented:
Looks good so far and passes all the tests.  When do you expect the release to happen?

lrineau committed:
Use PATCHES directly in vcpkg_from_github

lrineau commented:
[**@Rastaban**](https://github.com/Rastaban) commented on [Mar 19, 2019, 9:25 PM GMT+1](https://github.com/Microsoft/vcpkg/pull/5734#issuecomment-474567579 "2019-03-19T20:25:14Z - Replied by Github Reply Comments"):
> Looks good so far and passes all the tests.  When do you expect the release to happen?

Hopefully next week. Unfortunately, we discovered a bug in our CMake scripts in CGAL-4.14-beta1, and then this one in CGAL-4.14-beta2. The difficulty with bugs in CMake scripts is that our daily test suite cannot test all the possible usages (we try our best). That is why I wanted to test the beta releases with the downstream package managers (like vcpkg and brew). I am wondering if we should publish a third beta release.

Cc: @maxGimeno

lrineau committed:
CGAL-4.14-beta3

I do not plan to announce this beta release. It is only there to test
package managers.

lrineau committed:
CGAL-4.14-beta4

lrineau committed:
CGAL now depends also on Boost Multiprecision

maxGimeno committed:
Update to release 4.14

lrineau commented:
@Rastaban The final (non-beta) version of CGAL-4.14 has been published, and we have updated the PR. It is now ready to be evaluated and eventually merged.
